### PR TITLE
Fix false-flag of CompareExchange and re-usage with loops

### PR DESCRIPTION
**Pull request by Jozkee**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/6957

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs [hunk 1]
```diff
@@ -103,6 +103,12 @@ private static bool IsLocalUsedAsArgumentForJsonSerializerOnly(IObjectCreationOp
                     continue;
                 }
 
+                // Symbol is declared in a parent scope, so this implies that options are re-used.
```
Comment (eiriktsarpalis) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs:106]: For my education, would this also disable the diagnostic in something like this?
```C#
var options = new JsonSerializerOptions();
{
     JsonSerializer.Serialize(value, options);
}
```
If that's the case FWIW I don't think that's a problem we would need to address.

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs [hunk 2]
```diff
@@ -156,6 +162,26 @@ private static bool IsLocalUsedAsArgumentForJsonSerializerOnly(IObjectCreationOp
             return operation;
         }
 
+        private static bool IsLocalReferenceInsideChildLoop(ILocalReferenceOperation localRef, IBlockOperation symbolBlock)
+        {
+            IOperation? current = localRef;
+            while ((current = current?.Parent) is not null)
+            {
+                if (current is ILoopOperation loop)
```
Comment (eiriktsarpalis) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs:170]: I'm guessing this answers my earlier question in https://github.com/dotnet/roslyn-analyzers/pull/6979/files#r1358073471

Comment (Jozkee) [reply src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs:170]: I'll try to improve the comment.

**Review by eiriktsarpalis — APPROVED** [review]
Thanks @Jozkee!

**Comment by eiriktsarpalis** [comment]
@buyaa-n is there any action required to get this fix bundled with .NET 8?

**Comment by buyaa-n** [comment]
I think it's same as runtime porting, let managers know and agreed with porting, create a backport with `/backport to release/8.0`, fill in the description, I believe then need to send out emails to tactics for approval. CC @carlossanlop

**Pull request merged**